packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3445,25 +3472,61 @@ class _WordBoundary extends _TextBoundary {
 // interpreted as caret locations because [TextPainter.getLineAtOffset] is
 // text-affinity-aware.
 class _LineBreak extends _TextBoundary {
-  const _LineBreak(this.textLayout, this.textEditingValue);
+  const _LineBreak(
+    this.textLayout,
+    this.textEditingValue,
+    [this.continuesAtWrap = false]
```

Comment:
I think ideally this shouldn't be implemented in `_LineBreak`, as it's only supposed to tell you where the boundary is.

Reply:
I've moved it to the action and I think it makes more sense there, thanks for the suggestion.

In order to do this I added continuesAtWrap to DirectionalCaretMovementIntent even though most of its subclasses don't use continuesAtWrap.  Let me know if you think that's not alright.